tl.tl

```diff
@@ -4674,10 +4746,10 @@ parse_statements = function(ps: ParseState, i: integer, toplevel?: boolean): int
    end
 
    end_at(node, ps.tokens[i])
-   return i, node
+   return i, node, unattached_comments
```

Comment:
@upedd Actually, after looking at all the passing around of this `unattached_comments` value, I'm wondering if instead it wouldn't be simpler to store it as an attribute of the toplevel AST node.

I know that right now there is a single `Node` type for everything, but my plan is to eventually give nodes their own record types (like we do for `Type`), and than should sort things out in a cleaner way. Then we could even have a more specific node type for the toplevel. But for now, I wouldn't mind adding an `unattached_comments` attribute to the big `Node` record and have unattached comments from the toplevel get stored there, in the toplevel node.